### PR TITLE
Enable initial Thread Network Diagnostics cluster on the controllers

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
#### Problem
Thread Network Diagnostics isn't enable in the controllers

#### Change overview
Enable the Cluster on the controller .zap configuration file using the zap tool (`./scripts/tools/zap/configure.sh`)
and Generate the files with  `scripts/tools/zap_regen_all.py`

The configuration and code generation are seperated in two commits

**TODO**
the resetcounts command is optional, but even when enabled, zap do not generate the code for it because there is currently an issue where only one command of the same name is generate.

I could differ from the spec and rename the command or let it like this and regenerate the files once there is a fix in ZAP

I created the issue here. 
https://github.com/project-chip/zap/issues/161  

#### Testing
Rebuild chip-tool
./scripts/examples/gn_build_example.sh examples/chip-tool out/standalone
Test the cluster availability
./out/standalone/chip-tool threadnetworkdiagnostics
- Validate that all the commands are listed*
./out/standalone/chip-tool threadnetworkdiagnostics read
- Validate that all attributes are listed

### CODE REVIEW

File: src/darwin/Framework/CHIP/templates/helper.js [hunk 1]
```diff
@@ -16,47 +16,44 @@
  */
 
 // Import helpers from zap core
-const zapPath      = '../../../../../third_party/zap/repo/src-electron/';
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/templates/helper.js]: The random reindenting here is not great.....

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
`testSendClusterThreadNetworkDiagnosticsReadAttributeOperationalDatasetComponentsWithResponseHandler` is failing now that it's being tested...  Need to get that sorted out.

File: src/darwin/Framework/CHIP/templates/helper.js (hunk 1 above)
Comment (jmartinez-silabs) [reply src/darwin/Framework/CHIP/templates/helper.js]: This indent got fixed in the second commit when i ran the restyler. I this its my  vscode formating that screwed something there the first time when i did save. Should be ok now. 

And for the failing test I will take a look, Most of it is generated code so I am suprise it fails

File: src/app/zap-templates/zcl/thread-network-diagnostics-cluster.xml [hunk 2]
```diff
@@ -144,10 +144,12 @@ limitations under the License.
     <attribute side="server" code="0x39" define="PENDING_TIMESTAMP" type="INT64U" min="0x0000000000000000" max="0xFFFFFFFFFFFFFFFF" writable="false" default="0x0000000000000000" optional="true">PendingTimestamp</attribute>
     <attribute side="server" code="0x3A" define="DELAY" type="INT32U" min="0x00000000" max="0xFFFFFFFF" writable="false" default="0x0000" optional="true">delay</attribute>
     <!--TODO Attribute with struct as type is not currently supported. See attr 0x3B - SECURITY_POLICY and 0x3D OPERATIONAL_DATASET_COMPONENTS -->
-    <attribute side="server" code="0x3B" define="SECURITY_POLICY" type="ARRAY" entryType="SecurityPolicy" length="1" writable="false" optional="false">SecurityPolicy</attribute>
+    <!-- SECURITY_POLICY Lenght = 2 + (arraysize(1) * Size of Struct (3b))-->
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/thread-network-diagnostics-cluster.xml:147]:
```suggestion
    <!-- SECURITY_POLICY Length = 2 + (arraysize(1) * Size of Struct (3b))-->
```

File: src/app/zap-templates/zcl/thread-network-diagnostics-cluster.xml [hunk 3]
```diff
@@ -144,10 +144,12 @@ limitations under the License.
...
+    <attribute side="server" code="0x3B" define="SECURITY_POLICY" type="ARRAY" entryType="SecurityPolicy" length="5" writable="false" optional="false">SecurityPolicy</attribute>
     <attribute side="server" code="0x3C" define="DIAG_CHANNEL_MASK" type="INT8U" min="0x00" max="0xFF" writable="false" default="0x0000" optional="false">ChannelMask</attribute>
     <!-- Attribute with struct as type is not currently supported -->
-    <attribute side="server" code="0x3D" define="OPERATIONAL_DATASET_COMPONENTS" type="ARRAY" entryType="OperationalDatasetComponents" length="1" writable="false" optional="false">OperationalDatasetComponents</attribute>
+    <!-- OPERATIONAL_DATASET_COMPONENTS Lenght = 2 + (arraysize(1) * Size of Struct (12b))-->
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/thread-network-diagnostics-cluster.xml:151]:
```suggestion
    <!-- OPERATIONAL_DATASET_COMPONENTS Length = 2 + (arraysize(1) * Size of Struct (12b))-->
```

File: examples/all-clusters-app/all-clusters-common/gen/attribute-size.cpp [hunk 4]
```diff
@@ -761,6 +761,9 @@ uint16_t emberAfAttributeValueListSize(ClusterId clusterId, AttributeId attribut
     // The first 2 bytes specify the number of entries. A value of 0xFFFF means the list in invalid
     // and data is undefined.
     uint16_t entryCount = emberAfGetInt16u(buffer, 0, kSizeLengthInBytes);
+
+    ChipLogProgress(Zcl, "*DEBUG* Cluster 0x%04x,Attribute 0x%02x EntryCount 0x%04x", clusterId, attributeId, entryCount);
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/all-clusters-common/gen/attribute-size.cpp:765]: This shouldn't still be here.

File: examples/all-clusters-app/all-clusters-common/gen/attribute-size.cpp [hunk 5]
```diff
@@ -933,6 +936,8 @@ uint16_t emberAfAttributeValueListSize(ClusterId clusterId, AttributeId attribut
     }
 
     uint32_t totalSize = kSizeLengthInBytes + (entryCount * entryLength);
+    ChipLogProgress(Zcl, "*DEBUG* totalSize %d", totalSize);
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/all-clusters-common/gen/attribute-size.cpp:939]: This shouldn't still be here.

Comment (jmartinez-silabs) [reply examples/all-clusters-app/all-clusters-common/gen/attribute-size.cpp]: revert pushed

**Comment by bzbarsky-apple** [comment]
@jmartinez-silabs The tv clusters strike again.  You probably want to regenerate those to get rid of the unintended changes to them....

**Review by bzbarsky-apple — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Comment by jmartinez-silabs** [comment]
I will have to rebase and Regen due to conflicts but can I have a few more reviews

@andy31415 @Damian-Nordic @pan-apple @chrisdecenzo

**Review by Damian-Nordic — APPROVED** [review]

**Pull request merged**